packages/auth-remix/readme.md

```diff
@@ -0,0 +1,157 @@
+
+3. Now we just need to setup the UI to allow your users to sign in/up, etc. The easiest way to get started is to use the EdgeDB Auth's builtin UI. Or alternatively you can implement your own custom UI.
+
+   **Builtin UI**
+
+   To use the builtin auth UI, first you will need to enable the UI in the auth ext configuration (see the auth ext docs for details). For the `redirect_to` and `redirect_to_on_signup` configuration options, set them to `{your_app_url}/auth/builtin/callback` and `{your_app_url}/auth/builtin/callback?isSignUp=true` respectively. (Note: if you have setup the auth route handlers under a custom path, replace `auth` in the above url with that path).
+
+   Then you just need to configure the `onBuiltinUICallback` route handler to define what to do once the builtin ui redirects back to your app, and place a link to the builtin UI url returned by `auth.getBuiltinUIUrl()` somewhere in your app.
+
+   **Custom UI**
+
+   To help with implementing your own custom auth UI, the `auth` object has a number of methods you can use:
+
+   - `emailPasswordSignUp`
+   - `emailPasswordSignIn`
+   - `emailPasswordResendVerificationEmail`
+   - `emailPasswordSendPasswordResetEmail`
+   - `emailPasswordResetPassword`
+   - `signout`
+   - `isPasswordResetTokenValid(resetToken: string)`: Checks if a password reset token is still valid.
+   - `getOAuthUrl(providerName: string)`: This method takes the name of an OAuth provider (make sure you configure providers you need in the auth ext config first using CLI or EdgeDB UI) and returns a link that will initiate the OAuth sign in flow for that provider. You will also need to configure the `onOAuthCallback` auth route handler.
+
+## Usage
+
+Now you have auth all configured and user's can signin/signup/etc. you can use the `auth.getSession()` method in your app pages to retrieve an `AuthSession` object. This session object allows you to check if the user is currently logged in with the `isLoggedIn` method, and also provides a `Client` object automatically configured with the `ext::auth::client_token` global, so you can run queries using the `ext::auth::ClientTokenIdentity` of the currently signed in user.
+
+```ts
+// app/routes/_index.tsx
+import type { LoaderFunctionArgs } from "@remix-run/node";
+import { json } from "@remix-run/node";
+import { useLoaderData, Link } from "@remix-run/react";
+import { client, auth } from "~/services/auth.server";
+import { transformSearchParams } from "~/utils";
+
+export const loader = async ({ request }: LoaderFunctionArgs) => {
+  const builtinUIEnabled = await client.queryRequiredSingle<boolean>(
+    `select exists ext::auth::UIConfig`
+  );
+
+  const session = auth.getSession(request);
+  const isLoggedIn = await session.isLoggedIn();
+  const username = isLoggedIn
+    ? (await session.client.querySingle<string>(
+        `select global currentUser.name`
```

Comment:
I know this is just an example, but could it be confusing to use a custom global here? since `currentUser` isn't part of the auth ext schema, but looks like it could be?

Reply:
Also presumably in this case, checking `isLoggedIn` first would be redundant, as `currentUser` would be null if not signed in.